### PR TITLE
Swarm filters save config (fix #2878)

### LINKED ISSUE #2878: Filters added via command are not saved to config
I've got report on fc00 IRC that filters are not saved to config.
Need checking, fixing and adding test case.

### PR DESCRIPTION
**Pull request by yuvallanger**
Please review. (ref: #2878)

### CODE REVIEW

File: core/commands/swarm.go [hunk 1]
```diff
@@ -458,6 +461,23 @@ add your filters to the ipfs config file.
...
+		defer r.Close()
+		cfg, err := r.Config()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		if len(req.Arguments()) == 0 {
+			res.SetError(errors.New("no filters to add"), cmds.ErrClient)
+			return
```
Comment (Kubuxu) [inline core/commands/swarm.go]: Move this before opening the repo

**Comment by Kubuxu** [comment]
To prevent future regressions it would be great to have test cases for this behaviour, please check `test/sharness/t0141-addfilter.sh` and write testcase that will be testing fixed behaviour.

Apart from that it looks quite good.

**Comment by yuvallanger** [comment]
Added tests to `test/sharness/t0141-addfilter.sh`.
@Kubuxu LFCR

File: test/sharness/t0141-addfilter.sh [hunk 2]
```diff
@@ -31,38 +31,74 @@ test_swarm_filter_cmd() {
 	'
 }
 
+test_config_swarm_addrfilters_cmd() {
+	printf "" > list_expected
```
Comment (Kubuxu) [inline test/sharness/t0141-addfilter.sh]: You can use `touch` instead.

**Comment by yuvallanger** [comment]
`touch` does not truncates an existing file. Filter addresses are only added to the `list_expected` file with each call to `test_config_swarm_addrfilters_cmd`.

**Comment by Kubuxu** [comment]
LGTM

**Comment by whyrusleeping** [comment]
LGTM, thanks @yuvallanger

**Pull request merged**